src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -857,10 +857,18 @@ def get_system_mac(namespace=None, hostname=None):
             profile_cmd = ["false"]
             (mac, err) = run_command(profile_cmd)
         hw_mac_entry_outputs.append((mac, err))
-        (mac, err) = run_command(syseeprom_cmd)
-        hw_mac_entry_outputs.append((mac, err))
         (mac, err) = run_command_pipe(iplink_cmd0, iplink_cmd1, iplink_cmd2)
         hw_mac_entry_outputs.append((mac, err))
+        for (mac, err) in hw_mac_entry_outputs:
+            if err:
+                continue
+            mac = mac.strip()
+            if _valid_mac_address(mac):
+                return mac
+        # If mac not found, fetch from syseeprom
```

Comment:
Hi @anamehra , Look like the code logic between 862 and 867 is the same with 883 and 888,
but the return logic in 867 doesn't looks very clean because most platforms put mac addr into hw_mac_entry_outputs, this could cause we lost post-actions(like in line 890, it doesn't look to be an issue for now, but in the future it may do more transofrmation),
Let's change to put the valid mac into hw_mac_entry_outputs,
and on 869,870 ,assert whether hw_mac_entry_outputs is not empty, only when it's empty, we fetch mac from syseeprom

Reply:
Hi @yejianquan , we need to do validity check here and not just empty check. If valid not found, then only fetch syseeprom.
I can remove return and let the function return at the end but will need validity check to decide if syseeprom need to be used. Is that fine?